ios/authentication.md

```diff
@@ -1,344 +1,417 @@
+#### AWS Credentials
 
-    ```
-    ❯ Yes, use the default configuration.
-    ```
+AWS Credentials are used for signing requests to services that use AWS IAM, and for mobile clients they are provided by Amazon Cognito Identity Pools. Similar to JWT tokens, `AWSMobileClient` will return valid AWS Credentials from your cache immediately if they have not expired. If they are expired they will be refreshed using the JWT token that has been federated if the session is authenticated. For Guest scenarios they will be automatically refreshed. 
 
-3. When configuration for email and password sign-in is complete, a message appears confirming that you have configured local CLI metadata for this category. You can confirm this by viewing status.
+### Offline support
 
-    ```bash
-    $ amplify status
-      | Category | Resource name   | Operation | Provider plugin   |
-      | -------- | --------------- | --------- | ----------------- |
-      | Auth     | cognitoabcd0123 | Create    | awscloudformation |
-    ```
+`AWSMobileClient` is optimized to account for applications transitioning from offline to online connectivity, and refreshing credentials at the appropriate time so that errors do not occur when actions are taken and connectivity is not available. In no cases will the `AWSMobileClient` automatically sign out a user if connectivity is not available. You must always make an explicit `signOut()` call for a user to be signed out of a session. 
 
-4. To create your backend AWS resources run the following:
+In most cases if you are offline and make a service request, and your tokens are valid, the `AWSMobileClient` will pass the request directly to the service client. Therefore it is your responsibility to check network connectivity. In the case of the AWS AppSync client it supports offline operations and the request will be enqueued and automatically sent when connectivity is restored, refreshing credentials if necessary. [See the API guide for more information on AppSync](./api).
 
-    ```bash
-    $ amplify push
-    ```
+If you are offline and make a service request, and your tokens are **NOT** valid, the service request will be blocked and notifications for `signedOutUserPoolsTokenInvalid` or `signedOutFederatedTokensInvalid` will be sent to the listener. In the case of the AppSync client this can be ignored and the queries will come from cache or mutations enqueued with credentials automatically refreshing upon reconnection. For all other services, if this happens and you are offline you should not make the service request until you come back online, at which point the `AWSMobileClient` will automatically re-enter the token refresh flow outlined above and then make the service call with the updated credentials.
 
-5. Follow the [Set up Email & Password Login](set-up-email-and-password) steps to connect to your backend from your app.
+## Automated Setup
 
-#### Facebook
+Run the following command in your project's root folder:
```

Comment:
should we make a statement that we expect you amplify project to be initialized before running `amplify add auth` command?